### PR TITLE
Profile fixes

### PR DESCRIPTION
**Pull request by evertton**
- **Fixed update user language, when underline is a character in language selector**
  - When the language "Portuguese (Brazil)" was selected, the prevent LFI regular expression removed the underline character, so whenever the user chose this language they ended up receiving English as a fallback.

    ```php
    echo $inputData['language']; // portuguese_br
    
    // Prevent LFI.
    $inputData['language'] = preg_replace('/[^a-z]/', "", $inputData['language']);
    
    echo $inputData['language']; // portuguesebr
    ```

- **Fixes timezone selection display in profile**
  - The way the logic was before, the system timezone was always selected. When selecting "America/Maceio" timezone in the profile and saving, the next time it was displayed, "UTC" appeared again. It was necessary to choose the preferred timezone whenever making any edit.

### CODE REVIEW

File: pages/profile.php [hunk 1]
```diff
@@ -396,16 +396,15 @@
...
                                             <select class="form-control" id="profile-user-timezone">
                                                 <?php
                                                 foreach ($zones as $key => $zone) {
                                                     echo '
                                                 <option value="' . $key . '"',
-                                                    $session->has('user-timezone') && $session->get('user-timezone') && null !== $session->get('user-timezone') && $session->get('user-timezone') === $key ?
-                                                    ' selected' :
-                                                    (isset($SETTINGS['timezone']) === true && $SETTINGS['timezone'] === $key ? ' selected' : ''),
+                                                    $session->has('user-timezone') && $session->get('user-timezone') === $key ? ' selected' :
+                                                    (($session->has('user-timezone') && $session->get('user-timezone') === 'not_defined') && isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key ? ' selected' : ''),
```
Comment (corentin-soriano) [inline pages/profile.php:407]: It should work but I find that this ternary condition is complicated to read due to its length.

**Review by corentin-soriano — APPROVED** [review]

File: pages/profile.php (hunk 1 above)
Comment (nilsteampassnet) [reply pages/profile.php:407]: Can't this be replaced by

```
$userTimezone = $session->get('user-timezone');
$selected = ($userTimezone === $key) ? ' selected' : '';

if ($userTimezone === 'not_defined' && isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key) {
    $selected = ' selected';
}

return $selected;

```

Comment (evertton) [reply pages/profile.php:407]: I agree with you, is complicated to read. I would have liked to write the code like below, but I chose to stick with the standard adopted earlier.

```php
<?php foreach ($zones as $key => $zone): ?>
    <option value="<?php echo $key; ?>"<?php 
    if ($session->has('user-timezone'))
        if($session->get('user-timezone') === $key)
            echo ' selected';
        elseif ($session->get('user-timezone') === 'not_defined')
            if (isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key)
                echo ' selected';
    ?>><?php echo $zone; ?></option>
<?php endforeach; ?>
```

Should I commit the rewrite?

Comment (nilsteampassnet) [reply pages/profile.php:407]: Yes seems nice.

Can you please also sign your commit (see https://docs.github.com/fr/authentication/managing-commit-signature-verification/about-commit-signature-verification)

**Pull request merged**